wodles/aws/buckets_s3/waf.py

```diff
@@ -56,3 +84,32 @@ def json_event_generator(data):
                         sys.exit(9)
 
         return json.loads(json.dumps(content))
+
+    def get_service_prefix(self, account_id):
+        return AWSLogsBucket.get_service_prefix(self, account_id)
+        
+    def get_full_prefix(self, account_id, account_region, acl_name=None):
+        if self.type == WAF_NATIVE:
+            path = self.client.list_objects_v2(Bucket=self.bucket)
+            if 'Contents' in path:
+                for obj in path['Contents']:
+                    log_key = obj['Key']
+                    parts = log_key.split("/")
+                acl_name = parts[parts.index("WAFLogs") + 2]
```

Comment:
I need to better understand what this is for. Would it work if only the first object in `path['Contents']`? As it is now, it will still iterate all items inside `path['Contents']`, but this part:
```
acl_name = parts[parts.index("WAFLogs") + 2]
```
will be using only the `parts.index("WAFLogs")` of the last object found in the loop. Additionally, if `Contents` is empty, the loop will not iterate and `parts` would be undefined.

Reply:
## Update 

As we talked about in the huddle, with the new change, access the last  `Key` of `Contents` to obtain the `acl_name`